File: 2024/day01/python/day01.py
# 2024 Advent of Code: --- Day 1: Historian Hysteria ---

def open_input(location) -> list:
    input = open(location)
    content = input.readlines()
    input.close()
    return content

# part 1

a_list = []
b_list = []
content = open_input('../input.txt')

for line in content:
    a, b = line.split("   ")
    a = int(a.strip())
    b = int(b.strip())
    a_list.append(a)
    b_list.append(b)

a_list.sort()
b_list.sort()

i = 0
sum_dist = 0
while i < len(a_list):
    distance = abs(a_list[i] - b_list[i])
    sum_dist += distance
    i += 1

print("part I:", sum_dist)

# part 2

sim_score = 0

for num in b_list:
    if num in a_list:
        sim_score += num

print("Part II:", sim_score)

File: 2024/day01/cpp/day01.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int main() {
	ifstream input_file("../input.txt");
	
	vector<int> col1;
	vector<int> col2;
	int num1, num2;

	while (input_file >> num1 >> num2) {
		col1.push_back(num1);
		col2.push_back(num2);
	}
	input_file.close();

	sort(col1.begin(), col1.end());
	sort(col2.begin(), col2.end());
	
	int sum1 = 0;
	int sum2 = 0;
	for (int i = 0; i < col1.size(); i++) {
		sum1 += abs(col1[i] - col2[i]);
	}
	cout << "The sum distance of part I: \n";
	cout << sum1 << endl;

	for (int i = 0; i < col1.size(); i++) {
		for (int j = 0; j < col2.size(); j++) {
			if (col1[i] == col2[j]) {
				sum2 += col1[i];
			}
		}
	}

	cout << "The sum distance of part II: \n";
	cout << sum2 << endl;
	return 0;
}
